SwrveSDK/src/main/java/com/swrve/sdk/config/SwrveConfigBase.java

```diff
@@ -296,6 +329,158 @@ public SwrveConfigBase setDbName(String dbName) {
+     *
+     * @param eventOrContent
+     * @throws MalformedURLException
+     */
+    private void regenerateUrlFor(String eventOrContent) throws MalformedURLException {
+        Uri.Builder uriBuilder = new Uri.Builder();
+        String authority = "";
+        boolean generatingEventURL = eventOrContent.equalsIgnoreCase("event");
+        boolean generatingContentURL = eventOrContent.equalsIgnoreCase("content");
+
+        // Set the appID prefix but only for supported stacks
+        boolean isAppIdSupportedStack  = (stack == SwrveStack.US || stack == SwrveStack.EU || stack == SwrveStack.ACTIVISION);
+        if (null != appIdPrefix && isAppIdSupportedStack){
+            authority = Integer.toString(appIdPrefix) + ".";
+        }
+
+        // Host and Scheme to target
+        String scheme, endpoint;
+        if (generatingEventURL){
+            scheme = getUseHttpsForEventsUrl() ? "https" : "http";
+            endpoint = "api.swrve.com";
+        } else if(generatingContentURL) {
+            scheme = getUseHttpsForContentUrl() ? "https" : "http";
+            endpoint = "content.swrve.com";
+        }else{
+            SwrveLogger.wtf(LOG_TAG, "Unrecognized URL type asked to be generated :: " + eventOrContent);
+            return;
+        }
+
+        String prefix = getStackPrefix();
+
+        // Set the stack prefix and host
+        authority = authority + getStackPrefix() + endpoint;
+        Uri uri = uriBuilder.scheme(scheme).authority(authority).build();
+        URL newUrl = new URL(uri.toString());
+        if (generatingContentURL){
+            this.setContentUrl(newUrl, false);
+        }else if(generatingEventURL){
+            this.setEventsUrl(newUrl, false);
+        }
+    }
+
+
+    /**
+     * Simple helper method to setup the EU stack
+     *
+     * @return
+     */
+    public SwrveConfigBase useEuStack() throws MalformedURLException {
+        setStack(SwrveStack.EU);
+        regenerateURLS();
+        return this;
+    }
+
+    /**
+     * Set the stack being currently used by the client
+     * 
+     * @param stack
+     * @return
+     */
+    public SwrveConfigBase setStack(SwrveStack stack) throws MalformedURLException {
+        this.stack = stack;
+        regenerateURLS();
+        return this;
+    }
+
+    /**
+     * @return
+     */
+    public Integer getFeatureStackNum() {
+        return featureStackNum;
+    }
+
+    /**
+     *
+     * @param featureStackNum
+     * @return
+     */
+    public SwrveConfigBase setFeatureStackNum(Integer featureStackNum) {
+        this.featureStackNum = featureStackNum;
+        regenerateURLS();
+        return this;
+    }
+
+    /**
+     * @return
+     */
+    public Integer getAppIdPrefix() {
+        return appIdPrefix;
+    }
+
+    /**
+     *
+     * @param appIdPrefix
+     * @return
+     */
+    public SwrveConfigBase setAppIdPrefix(Integer appIdPrefix) {
```

Comment:
So if a customers wants to add an app id prefix to their URL EG `https://2132.eu.api.com`, how should they do it?